sdks/java/core/src/main/java/org/apache/beam/sdk/coders/SerializableCoder.java

```diff
@@ -57,6 +57,20 @@
     return new SerializableCoder<>(clazz, type);
   }
 
+  @Override
+  public boolean consistentWithEquals() {
+    return false;
+  }
+
+  /**
+   * The structural value of the object is the object itself. The {@link SerializableCoder} should
+   * be only used for objects with a proper {@link Object#equals} implementation.
+   */
+  @Override
+  public Object structuralValue(T value) {
+    return value;
```

Comment:
I'd like some sort of sanity check (or at minimum, a warning) when this coder is constructed if we've agreed that this is fine; e.g. check that the class the coder is constructed for (in the `of` method) implements equals

That should be roughly:

```
try {
  if (clazz.isInterface() || !clazz.equals(clazz.getMethod(equals, Object.class).getDeclaringClass())) {
    LOG.warn("Can't verify serialized elements of type {} have well defined equals method. This may produce incorrect results on some {}", clazz.getSimpleName(), PipelineRunner.class.getSimpleName());
  }
} catch (NoSuchMethodException e) {
  // All concrete classes have an equals method declared in their class hierarchy.
  throw new AssertionError(String.format("Concrete class %s has no equals method", clazz));
}
```

Reply:
Done.